ui/src/SoundWave.js

```diff
@@ -0,0 +1,70 @@
+export class SoundWave {
+
+    constructor(canvas, stream) {
+        this.canvas = canvas;
+        this.stream = stream;
+        this.analyser = undefined;
```

Comment:
Question: is there a conceptual difference between adding undefined & not assigning an initial value?